stl/inc/ranges

```diff
@@ -7656,6 +7662,400 @@ namespace ranges {
+
+    public:
+        zip_transform_view() noexcept(
+            is_nothrow_default_constructible_v<_Movable_box<_Func>>&& is_nothrow_default_constructible_v<_Inner_view>) =
+            default;
+
+        constexpr explicit zip_transform_view(_Func _Function_, _ViewTypes... _Views) noexcept(
+            is_nothrow_move_constructible_v<_Movable_box<_Func>>&&
+                is_nothrow_constructible_v<_Inner_view, remove_reference_t<_ViewTypes>&&...>) // strengthened
+            : _Function(in_place, _STD move(_Function_)), _Zip(_STD move(_Views)...) {}
+
+        constexpr auto begin() noexcept(noexcept(_Iterator<false>{*this, _Zip.begin()})) // strengthened
+        {
+            return _Iterator<false>{*this, _Zip.begin()};
+        }
+
+        constexpr auto begin() const noexcept(noexcept(_Iterator<true>{*this, _Zip.begin()})) // strengthened
+            requires _Enable_const_begin_end
+        {
+            return _Iterator<true>{*this, _Zip.begin()};
+        }
+
+        constexpr auto end() noexcept(_Is_end_noexcept<false>) // strengthened
+        {
+            if constexpr (common_range<_Inner_view>) {
+                return _Iterator<false>{*this, _Zip.end()};
+            } else {
+                return _Sentinel<false>{_Zip.end()};
+            }
+        }
+
+        constexpr auto end() const noexcept(_Is_end_noexcept<true>) // strengthened
+            requires _Enable_const_begin_end
+        {
+            if constexpr (common_range<const _Inner_view>) {
+                return _Iterator<true>{*this, _Zip.end()};
+            } else {
+                return _Sentinel<true>{_Zip.end()};
+            }
+        }
+
+        constexpr auto size() noexcept(noexcept(_Zip.size())) // strengthened
+            requires sized_range<_Inner_view>
+        {
+            return _Zip.size();
+        }
+
+        constexpr auto size() const noexcept(noexcept(_Zip.size())) // strengthened
+            requires sized_range<const _Inner_view>
+        {
+            return _Zip.size();
+        }
+    };
+
+    template <class _Func, class... _Ranges>
+    zip_transform_view(_Func, _Ranges&&...) -> zip_transform_view<_Func, views::all_t<_Ranges>...>;
+
+    namespace views {
+        struct _Zip_transform_fn {
+        private:
+            template <class _Func, class... _Types>
+            static constexpr bool _Is_invocation_noexcept = []() noexcept {
+                if constexpr (sizeof...(_Types) == 0) {
+                    using _Decayed_func = decay_t<_Func>;
+
+                    // We do the static_asserts here, rather than in the definition of operator(), so that they
+                    // show up earlier in compiler diagnostics.
+                    static_assert(move_constructible<_Decayed_func>,
+                        "ERROR: When using std::views::zip_transform without any views, the specified callable must "
+                        "be move constructible (see [range.zip.transform.overview]/2.1.1).");

```

Comment:
When the standard says "ill-formed" for a CPO invocation, it means "ill-formed in a SFINAE-friendly way", i.e., not `static_assert`.

Reply:
Ah, I see that now. Here's the relevant wording from the C++ Working Draft:

> When the types of `Args...` do not meet the customization point object's requirements, `T` shall not have a function call operator that participates in overload resolution.